File: js-app/src/state-management/folder/folder-content-item.js
/**
 * Folder content item.
 */
export class FolderContentItem {
  /**
   * @type {string}
   */
  type;

  /**
   * @type {string}
   */
  id;

  /**
   * @type {string}
   */
  name;

  /**
   * @type {string}
   */
  size;

  /**
   * @type {string}
   */
  parentId;

  /**
   * @param {string} type
   * @param {string} id
   * @param {string} name
   * @param {string} size
   * @param {string} parentId
   */
  constructor(type, id, name, size, parentId) {
    this.type = type;
    this.id = id;
    this.name = name;
    this.size = size;
    this.parentId = parentId;
    Object.freeze(this);
  }
}

File: services/src/main/java/com/teamdev/filehub/views/folder/FolderContentItem.java
package com.teamdev.filehub.views.folder;

import com.google.common.base.Objects;
import com.google.common.base.Preconditions;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/**
 * DTO with information about the folder content item.
 */
public class FolderContentItem {

    private final String type;
    private final String id;
    private final String parentId;
    private final String name;

    public FolderContentItem(@Nonnull String type,
                             @Nonnull String id,
                             @Nullable String parentId,
                             @Nonnull String name) {
        this.type = Preconditions.checkNotNull(type);
        this.id = Preconditions.checkNotNull(id);
        this.parentId = parentId;
        this.name = Preconditions.checkNotNull(name);
    }

    public String type() {
        return type;
    }

    public String id() {
        return id;
    }

    public String parentId() {
        return parentId;
    }

    public String name() {
        return name;
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(type, id, parentId, name);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        FolderContentItem item = (FolderContentItem) o;
        return Objects.equal(type, item.type) &&
                Objects.equal(id, item.id) &&
                Objects.equal(parentId, item.parentId) &&
                Objects.equal(name, item.name);
    }
}
